services/agit/agit.go

```diff
@@ -48,15 +48,15 @@ func ProcReceive(ctx context.Context, repo *repo_model.Repository, gitRepo *git.
 			continue
 		}
 
-		if !strings.HasPrefix(opts.RefFullNames[i], git.PullRequestPrefix) {
+		if !opts.RefFullNames[i].IsFor() {
 			results = append(results, private.HookProcReceiveRefResult{
 				IsNotMatched: true,
 				OriginalRef:  opts.RefFullNames[i],
 			})
 			continue
 		}
 
-		baseBranchName := opts.RefFullNames[i][len(git.PullRequestPrefix):]
+		baseBranchName := opts.RefFullNames[i].ForBranchName()
```

Comment:
When reading this line, I guess `AgitBranchName` is clearer 🤣

Reply:
The `refs/for/xx` is used by Gerrit first as a 3rd usage and then merged into git. So for now, both gerrit and agit will use it.